src/components/Articles.jsx

```diff
@@ -1,54 +1,50 @@
 import { useEffect, useState } from "react";
 import {getArticles, getUsers} from '../utils/api'
+import Sorter from "../components/Sorter"
+import { Link } from 'react-router-dom'
+
 const Articles = () => {
 const [articles, setArticles] = useState([])
 const [users, setUsers] = useState([])
 useEffect(() => {
-    getArticles().then((article) => {
-        
-        setArticles(article.articles)
-    })
-},[])
-useEffect(() => {
-    getUsers().then((users) => {
-       
-        setUsers(users.users)
-    })
+    getArticles().then((article) => setArticles(article.articles))
+    getUsers().then((users) => setUsers(users.users))
 },[])
 
-    return (
-        
 
+    return (
    <div>
+
+      <Sorter setArticles={setArticles} />
        <ul>
+           
 
        {articles.map((article) => {
            let image;
            for (let i = 0; i<users.length; i++) {
                if(article.author === users[i].username) {
                     image = users[i].avatar_url
                }
```

Comment:
Same with this. Always best practice to not do any filtering or looping through a response from the server. Have a look at all the request you can make from your server.